lib/Runtime/Library/MathLibrary.cpp

```diff
@@ -816,6 +816,23 @@ namespace Js
     {
         double result = 0;
 
+#ifdef POW_DOESNT_SUPPORT_NEGATIVE_INFINITY
+        if ( signbit( x ) && 0 == NumberUtilities::LuLoDbl( x ) && 0 == ( NumberUtilities::LuHiDbl( x ) & 0x7FFFFFFF ) )
```

Comment:
@obastemur.  I think @digitalinfinity  means you should refactor the -0.0 detect into NumberUtilities.    But think some thing similar already exists.  See JavascriptNumber::IsNegZero.  That one is an instance method, so we can't really use it, but it is implemented as calling `NumberUtilities::IsSpecial(value, k_NegZero)`

So either provide a static version of NumberUtilties::IsNegZero or just directly call `NumberUtilities::IsSpecial(value, NumberUtilities::k_NegZero)` here to detect negative zero